config/prometheus/collector_alerts.yaml

```diff
@@ -55,6 +55,36 @@ spec:
       labels:
         service: collector
         severity: Warning
+    - alert: ElasticsearchDeprecation
+      annotations:
+        message: "The OpenShift Elasticsearch Operator is deprecated and is planned to be removed in a future release. Red Hat provides bug fixes and support for this feature during the current release lifecycle, but this feature no longer receives enhancements. As an alternative to using the OpenShift Elasticsearch Operator to manage the default log storage, you can use the Loki Operator."
+        summary: "Detected Elasticsearch as the on cluster storage which is being deprecated in a future release."
+      expr: |
+        sum(min_over_time(container_cpu_usage_seconds_total{container=~'elasticsearch'}[5m])) > 0
```

Comment:
This query seems to work, but is a bit "unspecific" in my opinion. We're not really interested in the CPU usage, only the existence of the elasticsearch pods. Also we're not interested in `elasticsearch` containers that belong to other applications (possibly of the customer).

I would suggest adding matchers to "drill down" into the pod that we want to monitor and also use a metric that just gives us "existence", for example one of the metrics provided by kube-state-metrics:

```suggestion
        sum(kube_pod_labels{namespace="openshift-logging", label_component="elasticsearch"}) > 0
```

Alternatively counting containers called "elasticsearch" similar to the previous query:

```suggestion
        sum(kube_pod_container_info{namespace="openshift-logging", container="elasticsearch"}) > 0
```

This suggestion also avoids the "time-based aggregation" and "range vector" which might cause flapping of the alert if the CPU usage is variable (which is probably no issue for ES but might be for Kibana). This should also mean that you can reduce the `for` time to five minutes or even less, because we do not have to wait for the result to "settle" because there is no time component involved.

Reply:
Incorporated the suggested metrics for the alerts. Thanks!